crates/floresta-chain/src/pruned_utreexo/consensus.rs

```diff
@@ -102,6 +98,82 @@ impl Consensus {
         sha256::Hash::from_slice(leaf_hash.as_slice())
             .expect("parent_hash: Engines shouldn't be Err")
     }
+
+    /// Check whether a block is valid
+    ///
+    /// This function is divided into two parts:
+    ///
+    /// Header validation with [`Consensus::validate_header`]
+    ///
+    /// Transaction validation with [`Consensus::validate_block_transactions`]
+    pub fn validate_block(
+        &self,
+        block: &bitcoin::Block,
+        ancestor: &BlockHeader,
+        height: u32,
+        utxo_data: UtxoMap,
+        flags: c_uint,
+        validate_script: bool,
+    ) -> Result<(), BlockchainError> {
+        //Check if the block declared merkle root is the same as the one calculated.
+        if !block.check_merkle_root() {
+            return Err(BlockchainError::BlockValidation(
+                BlockValidationErrors::BadMerkleRoot,
+            ));
+        }
+
+        //Check bip34 activation and presence
+        if height >= self.parameters.bip34_activation_height
+            && block.bip34_block_height() != Ok(height as u64)
+        {
+            return Err(BlockchainError::BlockValidation(
+                BlockValidationErrors::BadBip34,
+            ));
+        }
+
+        //Check if the block witness commitment is valid
+        if !block.check_witness_commitment() {
+            return Err(BlockchainError::BlockValidation(
+                BlockValidationErrors::BadWitnessCommitment,
+            ));
+        }
+
+        //Check if the block version is valid accordingly with the height
+        //and bip activation.
+        if !self.verify_block_version(height, block.header.version.to_consensus()) {
+            return Err(BlockchainError::BlockValidation(
+                BlockValidationErrors::BadBlockVersion,
+            ));
+        }
+
+        //Here is important to check that the declared previous block hash is
+        //the same as the actual previous block hash from the chain.
+        if block.header.prev_blockhash != ancestor.block_hash() {
+            return Err(BlockchainError::BlockValidation(
+                BlockValidationErrors::BlockExtendsAnOrphanChain,
+            ));
+        }
+
+        #[cfg(not(feature = "bitcoinconsensus"))]
+        let flags = 0;
+        self.verify_block_transactions(height, utxo_data, &block.txdata, flags, validate_script)?;
+        Ok(())
+    }
+
+    const fn verify_block_version(&self, height: u32, version: i32) -> bool {
+        return match version {
+            0 => false,
+            1 => height <= self.parameters.bip34_activation_height,
+            2 => height <= self.parameters.bip34_activation_height,
```

Comment:
This is why it's crashing. This should be `>=`. Right now, after the activation this will return false.

Reply:
Makes a lot of sense... but still strange that i couldnt replicate this